internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -167,47 +167,114 @@ func (cf CloudFormation) describeStack(describeStackInput *cloudformation.Descri
 func (cf CloudFormation) describeStackWithClient(describeStackInput *cloudformation.DescribeStacksInput,
 	client cloudformationiface.CloudFormationAPI) (*cloudformation.Stack, error) {
 	describeStackOutput, err := client.DescribeStacks(describeStackInput)
-	if err != nil {
+	stackNotFoundError := stackDoesNotExist(err)
```

Comment:
Does this work if `err = nil`?
I find this to be slightly easier to reason about:
```go
if err != nil {
   if stackDoesNotExist(err) {
       return nil, &ErrStackNotFound{stackName: *describeStackInput.StackName}
   }
   return nil, err
}
if len(describeStackOutput.Stacks) == 0 {
   return nil, &ErrStackNotFound{stackName: *describeStackInput.StackName}
}
```

Reply:
I like this - thanks Efe!